src/header/common/mod.rs

```diff
@@ -190,6 +214,8 @@ macro_rules! header {
                 self.fmt_header(f)
             }
         }
+
+        generate_header_serialization!($id);
```

Comment:
Unfortunately this doesn't work without exporting `generate_header_serialization` too, since the expansion is in another namespace. 

We can just export it with a `__` prefix like `__hyper_deref`.